File: subin/programmers/주식가격.js
function solution(prices) {
  let ans = Array(prices.length).fill(0)
  let stack = []
  for (let i = 0; i < prices; i++) {
    // 배열이 빈 값이 아니고 해당 가격이 이전에 넣었던 가격보다 적으면 ans에 넣기
    while (stack.length > 0 && prices[i] < prices[stack[stack.length]]) {
      x = stack.pop()
      ans[x] = i - x
    }
    stack.push(i)
  }
  // 끝까지 살아남은 자들의 값을 계산해주기!
  while (stack.length > 0) {
    y = stack.pop()
    ans[y] = prices.length - 1 - j
  }

  return ans
}
console.log(solution([1, 2, 3, 2, 3]))

File: sejeong/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0] * n
    stack = []  # 계산덜된 주식가격 인덱스 넣는 스택

    for i in range(n):
        while stack and prices[stack[-1]] > prices[i]:  # 전의 주식가격이 더크면 계산하기
            j = stack.pop()
            answer[j] = i - j
        stack.append(i)

    while stack: # 나머지 안떨어진 주식가격 계산하기
        j = stack.pop()
        answer[j] = n - 1 - j

    return answer

prices = [1, 2, 3, 2, 3]
print(solution(prices))
